File: client/src/app/layouts/training/model/trainee.ts
import { User } from 'src/app/views/pages/login/model/user.model';
import { EmpCode } from 'src/app/views/pages/model/emp-code';
import { BatchMaster } from './batch-master';

export class Trainee {
  traineeId: number;
  empCode: EmpCode;
  batchId: BatchMaster;
  createdBy: number;
  createdOn: Date;
  updatedBy: number;
  updatedOn: Date;
}

File: Java_Backend/src/main/java/com/netlink/rsk/model/Trainee.java
package com.netlink.rsk.model;

import lombok.Data;

import javax.persistence.*;
import java.util.Date;

@Data
@Entity
@Table(name = "Trainee", catalog = "Training")
public class Trainee {
    @Id
    @Column(name = "Traineeid", nullable=false)
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long traineeId;

    @OneToOne
    @JoinColumn(name = "Empcode", referencedColumnName = "Employeecode")
    private EmployeeMaster empCode;

    @OneToOne
    @JoinColumn(name = "Batchid", referencedColumnName = "Batchid")
    private BatchMaster batchId;

    @Column(name = "Createdby")
    private Integer createdBy;

    @Column(name = "Createdon")
    private Date createdOn;

    @Column(name = "Updatedby")
    private Long updatedBy;

    @Column(name = "Updatedon")
    private Date updatedOn;

}
